pkg/ipmi/ipmi.go

```diff
@@ -278,27 +265,41 @@ func (i *IPMI) RawSendRecv(msg Msg) ([]byte, error) {
 		msg:     recvMsg,
 	}
 
-	t := time.After(timeout)
-	for {
-		if err := ioctlGetRecv(i.Fd(), _IPMICTL_RECEIVE_MSG, recv); err != nil {
-			return nil, err
+	// This will be passed to the RawConn Read function. Until readMsg returns
+	// true, Read blocks and retries until the connection is ready for reading.
+	readMsg := func(fd uintptr) bool {
+		if err := ioctlGetRecv(fd, _IPMICTL_RECEIVE_MSG, recv); err != nil {
```

Comment:
Hm. I'm trying to figure out how to get the timeout back in here. This might be... annoying and we might have to go back to the old Select version of this. I hadn't thought of that.

Reply:
Actually, you can use os.File.SetReadDeadline on this. Just set the deadline before calling the syscall conn read.

Reply:
Cool, done!